allows user to get a single article

MFStevenson opened the pull request:
user can search by id and see article details
refactored to put state in app instead of wrapper as unsure how to conditionally render based on url

MFStevenson committed:
allows user to get a single article
user can search by id and see article details
refactored to put state in app instead of wrapper as unsure how to conditionally render based on url

nc-modou commented on src/components/ArticleStateWrapper.jsx:
```diff
@@ -4,24 +4,14 @@ import Articles from "./Articles";
-      <Articles articles={articles} />
-    </>
-  );
+  
+
+  console.log(window.location.pathname);
+
+  
```
Remember to take out console logs when you submit a PR

nc-modou commented on src/components/Articles.jsx:
```diff
@@ -1,10 +1,36 @@
   return (
     <>
       <h2>Articles</h2>
+      <form onSubmit={handleSubmit}>
+        <label>
+          Search Article ID:{" "}
+          <input
+            type="number"
+            placeholder="Article ID"
+            value={input}
+            onChange={updateInput}
+          ></input>
+        </label>
+        <button id="search-article-btn">Search</button>
+      </form>
```
The article ID means nothing to the user - they don't know which ID is which article. It would be much better to let them get to the article by linking to it from the homepage

nc-modou reviewed (commented):
Your single article page looks good - but the way that you're getting to it is unusual. All of our pages have Routes in the app.js, and to make a parametric endpoint, we use syntax like this path="articles/:articleID" and let the element for that get to the articlePage.

Then, on the page with all articles, we can make them link to the correct article id using the <Link> tag going to the correct url for each link. You may use this to allow to user to click 'view' and be taken to the right article, or turn the title into a link

This is how we should get to the single article pages, instead of having that saved in state. Please implement this change then submit another PR.

This will be useful to show you how to link properly
https://notes.northcoders.com/courses/js-front-end/react-router

MFStevenson committed:
actions feedback
created a new homepage which allows the user to either navigate to articles or topics which allows a /articles route to be added
topics page has yet to be implemented so button functionality has not been implemented yet
view button on each article now works

nc-poppy-m reviewed (approved):
Well done on actioning this feedback! Merge when ready
